presto-verifier/src/main/java/com/facebook/presto/verifier/framework/ExtendedVerification.java

```diff
@@ -134,8 +161,42 @@ public DataMatchResult verify(
                 return partitionMatchResult;
             }
         }
+        return null;
+    }
 
-        return dataMatchResult;
+    private DataMatchResult verifyBucket(QueryObjectBundle control, QueryObjectBundle test, List<Column> controlColumns, List<Column> testColumns, ChecksumQueryContext controlChecksumQueryContext, ChecksumQueryContext testChecksumQueryContext)
+    {
+        List<ChecksumResult> controlBucketChecksum = null;
+        List<ChecksumResult> testBucketChecksum = null;
+        try {
+            controlBucketChecksum = runBucketChecksum(control, controlColumns, controlChecksumQueryContext, CONTROL_BUCKET_CHECKSUM);
+            testBucketChecksum = runBucketChecksum(test, testColumns, testChecksumQueryContext, TEST_BUCKET_CHECKSUM);
+        }
+        catch (Throwable e) {
+            return null;
+        }
+        if (controlBucketChecksum.size() != testBucketChecksum.size()) {
+            return new DataMatchResult(
+                    BUCKET_DATA,
+                    BUCKET_COUNT_MISMATCH,
+                    Optional.empty(),
+                    OptionalLong.of(controlBucketChecksum.size()),
+                    OptionalLong.of(testBucketChecksum.size()),
+                    ImmutableList.of());
+        }
+        for (int i = 0; i < controlBucketChecksum.size(); i++) {
+            DataMatchResult bucketMatchResult = match(
+                    BUCKET_DATA,
+                    checksumValidator,
+                    controlColumns,
+                    testColumns,
+                    controlBucketChecksum.get(i),
+                    testBucketChecksum.get(i));
+            if (!bucketMatchResult.isMatched()) {
+                return bucketMatchResult;
+            }
+        }
+        return null;
```

Comment:
Returning `null` is what we try to avoid in this codebase, ex., we try to use Optional<>. 

On the other hand, in this case, are you able to return a "matched" DataMatchResult?